src/Model.ts

```diff
@@ -210,7 +200,6 @@ export default class Model extends Base {
       .then(data => {
         this.set(data)
         this.commitChanges()
-        return data
```

Comment:
any reason for this change?

Reply:
There's no reason for it unless there's a concatenated `then` statement.

```javascript
const promise = new Promise((resolve, reject) => { resolve('payload') })

promise.then(payload => { return 'payload1' })
promise.then(payload => console.log(payload))

// output => 'payload'
```

Reply:
I'm not sure I understand the question. Do we want to keep the promise returning the response or not? This is a backward incompatible change I understand.

Reply:
It is not a backward incompatible change. The promise resolves with the HTTP response to all its hooked `then` statements, regardless of the return statements of each particular `then` callback.

This is what I'm trying to explain in the piece of code above ☝️ , notice that the second `then` statement receives the original promise resolution result, and not the returned value of the first `then` statement.

Reply:
I think I see. I'm not sure the complexity added with `withRequest` is worth it. I think this is actually also the reason why the switch from `async/await` to raw promises (can't attach twice a `then` just using `await`).